app-backend/tile/src/main/scala/LayerCache.scala

```diff
@@ -146,4 +135,69 @@ object LayerCache extends Config with LazyLogging {
         }
       }
     }
+
+  /** Calculate the histogram for the least resolute zoom level to automatically render tiles */
+  def modelLayerGlobalHistogram(
+    ast: MapAlgebraAST,
+    params: EvalParams,
+    toolRun: ToolRun,
+    user: User
+  ): OptionT[Future, Histogram[Double]] =
+    histogramCache.cachingOptionT(s"model-${toolRun.id}-${ast.id}") { implicit ec =>
+      for {
+        lztile <- OptionT(Interpreter.interpretGlobal(ast, params.sources, TileSources.cachedGlobalSource).map(_.toOption))
+        tile   <- OptionT.fromOption[Future](lztile.evaluateDouble)
+      } yield {
+        val hist = StreamingHistogram.fromTile(tile)
+        val currentMetadata = params.metadata.getOrElse(ast.id, NodeMetadata())
+        val updatedMetadata = currentMetadata.copy(histogram = Some(hist))
+        val updatedParams = params.copy(metadata=params.metadata + (ast.id -> updatedMetadata))
+        val updatedToolRun = toolRun.copy(executionParameters=updatedParams.asJson)
+        database.db.run { ToolRuns.updateToolRun(updatedToolRun, toolRun.id, user) }
```

Comment:
This is a `Future`, but is ran outside the scope of the local `OptionT`. It might be safer (in terms of accounting for all behaviour) to rework this into a `flatMap` structure.